plugins/core/gs-appstream.c

```diff
@@ -828,6 +828,34 @@ gs_appstream_refine_app (GsPlugin *plugin,
 	return TRUE;
 }
 
+static gboolean
+matches_all_on_metadata (AsApp  *item,
+			 gchar **values)
+{
+	GHashTable *metadata = as_app_get_metadata (item);
+	gchar **value_iter = values;
+
+	for (; *value_iter != NULL; ++value_iter) {
+		GHashTableIter iter;
+		gboolean matched_value = FALSE;
+		gpointer key, value;
+
+		g_hash_table_iter_init (&iter, metadata);
+
+		while (g_hash_table_iter_next (&iter, &key, &value)) {
+			if (g_ascii_strcasecmp ((gchar *) key, *value_iter) == 0) {
```

Comment:
Why can't you lookup the key directly? Do we really need to use case insensitive match?

Reply:
We do - G-S lowercases all the search terms